mdata/cache/ccache_metric.go

```diff
@@ -202,13 +212,13 @@ func (mc *CCacheMetric) searchBackward(from, until uint32, keys []uint32, res *C
 	}
 
 	for ; ts != 0; ts = mc.chunks[ts].Prev {
+		if ts < from {
+			break
+		}
+
 		log.Debug("CCacheMetric searchBackward: backward search adds chunk ts %d to end", ts)
 		res.End = append(res.End, mc.chunks[ts].Itgen)
 		res.Until = ts
-
-		if ts <= from {
-			break
-		}
```

Comment:
I can well imagine that this is the reason for the error that @woodsaj saw. because if there is a disconnect in the linked list then it would seek to the disconnect from both sides, but the backwards one would seek one chunk too far, leading to the cassandra query receiving a `to` that's before the `from`.